lib/calendar.ts

```diff
@@ -445,7 +443,7 @@ impl['iso8601'] = {
     if (!HasSlot(date, ISO_YEAR)) date = ES.ToTemporalDate(date);
     return ES.LeapYear(GetSlot(date, ISO_YEAR));
   }
-};
+} as Omit<NonIsoImpl, 'helper'>;
```

Comment:
I vaguely recall being able to create these implementations using something like the following (untested):

```
impl['iso8601'] = (new (class implements CalendarImpl, Omit<NonIsoImpl, 'helper'> {
  .. impl here ...
})());
```

which might help TS better understand the inheritance / 'this' values more? unsure.